src/js/stores/DCOSStore.js

```diff
@@ -26,50 +27,42 @@ class DCOSStore extends EventEmitter {
 
     this.data = {
       marathon: new ServiceTree(),
-      mesos: {
-        frameworks: []
-      },
+      mesos: new SummaryList(),
       dataProcessed: false
     };
 
     this.proxyListeners = [
       {
         event: MARATHON_GROUPS_CHANGE,
-        handler: this.processMarathonData,
+        handler: this.onMarathonGroupsChange,
         store: MarathonStore
       },
       {
         event: MESOS_SUMMARY_CHANGE,
-        handler: this.processMesosData,
+        handler: this.onMesosSummaryChange,
         store: MesosSummaryStore
       }
     ];
   }
 
-  /**
-   * Process Marathon data
-   * @param {ServiceTree} data
-   */
-  processMarathonData(data) {
-    if (!(data instanceof ServiceTree)) {
+  onMarathonGroupsChange() {
+    let groups = MarathonStore.get('groups');
+    if (!(groups instanceof ServiceTree)) {
       return;
     }
 
-    this.data.marathon = data;
+    this.data.marathon = groups;
     this.data.dataProcessed = true;
     this.emit(DCOS_CHANGE);
   }
 
-  /**
-   * Process Mesos data
-   * @param {{frameworks:array}} data
-   */
-  processMesosData(data) {
-    if (data == null || !Array.isArray(data.frameworks)) {
+  onMesosSummaryChange() {
+    let states = MesosSummaryStore.get('states');
+    if (!(states instanceof SummaryList)) {
       return;
     }
-    this.data.mesos = data;
-    this.data.dataProcessed = true;
```

Comment:
Why is this not needed anymore? And not sure why it was set in both `processMarathonData` and `processMesosData` before either, but is it sufficient to have it in one place?